apps/server/src/modules/learnroom/service/course-copy.service.spec.ts

```diff
@@ -310,6 +336,78 @@ describe('course copy service', () => {
 
 			expect(courseCopy.color).toEqual(course.color);
 		});
+
+		it('should find all ctl tools for this course', async () => {
+			const { course, user } = setup();
+			await service.copyCourse({ userId: user.id, courseId: course.id });
+
+			expect(contextExternalToolService.findAllByContext).toHaveBeenCalledWith({
+				id: course.id,
+				type: ToolContextType.COURSE,
+			});
+		});
+
+		it('should copy all ctl tools', async () => {
+			const { course, user, tools } = setup();
+			const status = await service.copyCourse({ userId: user.id, courseId: course.id });
+			const courseCopy = status.copyEntity as Course;
+
+			expect(contextExternalToolService.copyContextExternalTool).toHaveBeenCalledWith(tools[0], courseCopy.id);
+			expect(contextExternalToolService.copyContextExternalTool).toHaveBeenCalledWith(tools[1], courseCopy.id);
+		});
+	});
+
+	describe('when FEATURE_CTL_TOOLS_COPY_ENABLED is false', () => {
+		const setup = () => {
+			const user = userFactory.buildWithId();
+			const allCourses = courseFactory.buildList(3, { teachers: [user] });
+			const course = allCourses[0];
+			const originalBoard = boardFactory.build({ course });
+
+			userRepo.findById.mockResolvedValue(user);
+			courseRepo.findById.mockResolvedValue(course);
+			courseRepo.findAllByUserId.mockResolvedValue([allCourses, allCourses.length]);
+			boardRepo.findByCourseId.mockResolvedValue(originalBoard);
+			roomsService.updateBoard.mockResolvedValue(originalBoard);
+
+			const courseCopyName = 'Copy';
+			copyHelperService.deriveCopyName.mockReturnValue(courseCopyName);
+			copyHelperService.deriveStatusFromElements.mockReturnValue(CopyStatusEnum.SUCCESS);
+
+			const boardCopyStatus = {
+				title: 'boardCopy',
+				type: CopyElementType.BOARD,
+				status: CopyStatusEnum.SUCCESS,
+				copyEntity: boardFactory.build(),
+				elements: [],
+			};
+			boardCopyService.copyBoard.mockResolvedValue(boardCopyStatus);
+
+			lessonCopyService.updateCopiedEmbeddedTasks.mockReturnValue(boardCopyStatus);
+
+			toolFeatures.ctlToolsCopyEnabled = false;
+
+			return {
+				user,
+				course,
+			};
+		};
+
+		it('should not find ctl tools', async () => {
+			Configuration.set('FEATURE_CTL_TOOLS_COPY_ENABLED', false);
```

Comment:
You dont need those line Configuration.set('FEATURE_CTL_TOOLS_COPY_ENABLED', false);